src/dmd/dcast.d

```diff
@@ -2801,10 +2842,10 @@ bool typeMerge(Scope* sc, TOK op, Type* pt, Expression* pe1, Expression* pe2)
         }
     }
 
-    Type t1 = e1.type;
-    Type t2 = e2.type;
```

Comment:
These are now redundant.

Reply:
there is this change
https://github.com/dlang/dmd/blob/5b3a28416610535a4d415e0f27323ed64ea3349f/src/dmd/dcast.d#L2840-L2841